include/envoy/secret/secret.h

```diff
@@ -0,0 +1,59 @@
+#pragma once
+
+#include <stdexcept>
+#include <string>
+#include <unordered_map>
+#include <vector>
+
+#include "envoy/common/exception.h"
+#include "envoy/ssl/context.h"
+
+namespace Envoy {
+namespace Secret {
+
+/**
+ * Secret contains certificate chain and private key
+ */
+class Secret {
+public:
+  virtual ~Secret() {}
+
+  /**
+   * @return a name of the SDS secret
+   */
+  virtual const std::string& getName() PURE;
+
+  /**
+   * @return a string of certificate chain
+   */
+  virtual const std::string& getCertificateChain() PURE;
+  /**
+   * @return a string of private key
+   */
+  virtual const std::string& getPrivateKey() PURE;
+};
+
+typedef std::shared_ptr<Secret> SecretSharedPtr;
+
+typedef std::unordered_map<std::string, SecretSharedPtr> SecretSharedPtrMap;
```

Comment:
The map and vector are not used in interfaces, so move them to impl.h?

Reply:
Moved,